core/server/api/posts.js

```diff
@@ -55,7 +55,27 @@ posts = {
          * @returns {Object} options
          */
         function modelQuery(options) {
-            return dataProvider.Post.findPage(options);
+            var requestedColumns = [];
+
+            // 'url' and 'author' depend on other columns. Make sure to load them.
+            if (_.isArray(options.columns)) {
+                requestedColumns = _.clone(options.columns);
+                if (_.includes(options.columns, 'url') || _.includes(options.columns, 'author')) {
+                    options.columns.push('id', 'published_at', 'slug', 'author_id');
```

Comment:
Just to make sure we are on the same page... I think we only need these 4 extra properties when `url` or `author` is present in the `fields` parameter. 
Do we treat them as default columns to fetch if at least 1 column is requested?